pkg/client/drive_adapter_test.go

```diff
@@ -0,0 +1,693 @@
+		},
+		{
+			backendVersion: "vbeta1",
+			kind:           "DirectCSIDrive",
+			name:           "drive-2",
+			newDriveName:   "newdrive-2",
+			driveRepresentation: &directcsi.DirectCSIDrive{
+				TypeMeta: utils.DirectCSIDriveTypeMeta(),
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "newdrive-2",
+				},
+			},
+		},
+		{
+			backendVersion: "vbeta3",
+			kind:           "DirectCSIDrive",
+			name:           "drive-3",
+			newDriveName:   "newdrive-3",
+			driveRepresentation: &directcsi.DirectCSIDrive{
+				TypeMeta: utils.DirectCSIDriveTypeMeta(),
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "newdrive-3",
+				},
+			},
+		},
+	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	for i, testCase := range testCases {
+		inter, err := getFakeDirectCSIAdapter(testCase.backendVersion, "directcsidrives", newUnstructured(testCase.backendVersion, testCase.kind, testCase.name))
+		if err != nil {
+			t.Errorf("case %v: Error in creating fake client, Expected err to nil, got %v", i+1, err)
+		}
+		fakeDirectCSIDriveAdapter := &directCSIDriveInterface{*inter}
+		_, err = fakeDirectCSIDriveAdapter.Create(ctx, testCase.driveRepresentation, metav1.CreateOptions{})
+		if err != nil {
+			t.Errorf("case %v: Error in Create, Expected  err to nil, got %v", i+1, err)
+		}
+
+		_, err = fakeDirectCSIDriveAdapter.Get(ctx, testCase.newDriveName, metav1.GetOptions{})
+		if err != nil {
+			t.Errorf("case %v: Error while fetching newly created drive, Expected  err to nil, got %v", i+1, err)
+		}
+	}
+}
+
+func TestDeleteDrive(t *testing.T) {
+	testCases := []struct {
+		backendVersion string
+		kind           string
+		name           string
+	}{
+		{
+			backendVersion: "v1alpha1",
+			kind:           "DirectCSIDrive",
+			name:           "direct-1",
+		},
+		{
+			backendVersion: "v1beta1",
+			kind:           "DirectCSIDrive",
+			name:           "drive-2",
+		},
+		{
+			backendVersion: "v1beta2",
+			kind:           "DirectCSIDrive",
+			name:           "drive-3",
+		},
+	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	for i, testCase := range testCases {
+
+		inter, err := getFakeDirectCSIAdapter(testCase.backendVersion, "directcsidrives", newUnstructured(testCase.backendVersion, testCase.kind, testCase.name))
+		if err != nil {
+			t.Errorf("case %v: Error in creating fake client, Expected err to nil, got %v", i+1, err)
+		}
+		fakeDirectCSIDriveAdapter := &directCSIDriveInterface{*inter}
+
+		err = fakeDirectCSIDriveAdapter.Delete(ctx, testCase.name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("case %v: Error in Delete, Expected  err to nil, got %v", i+1, err)
+		}
+		_, err = fakeDirectCSIDriveAdapter.Get(ctx, testCase.name, metav1.GetOptions{})
+
+		if err != nil && err.Error() != fmt.Sprintf("%s %q %s", "directcsidrives.direct.csi.min.io", testCase.name, "not found") {
+			t.Errorf("case %v: Error in Get after delete, Expected  err to nil, got %v", i+1, err)
+		}
```

Comment:
We can use https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound to check for not found errors